web/modules/custom/dpl_instant_loan/src/Form/DplInstantLoanSettingsForm.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\dpl_instant_loan\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Instant Loan settings form.
+ */
+class DplInstantLoanSettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames(): array {
+    return [
+      'dpl_instant_loan.settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'dpl_instant_loan_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('dpl_instant_loan.settings');
+
+    $form['enabled'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enabled', [], ['context' => 'dpl_instant_loan']),
+      '#description' => $this->t(
+        'Should the instant loan functionality be activated?',
+        [],
+        ['context' => 'dpl_instant_loan']
+      ),
+      '#default_value' => $config->get('enabled'),
+    ];
+
+    $form['match_string'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Match String', [], ['context' => 'dpl_instant_loan']),
```

Comment:
Consider if we have a dependency between fields here.

How will the React app work if `match_string` and/or `threshold` are empty if the feature is enabled? Should they in this case be required?

Reply:
Added state handling in the form. 👍